### ISSUE TITLE
Check and update architecture documentation

### ISSUE THREAD

**Issue by op-hub** [amosproj/amos2024ss04-building-information-enhancer#285]
## Description
Check documentation for completeness

## Acceptance criteria
* Update documentation (new API, etc)

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

**Comment by Corgam** [comment]
Updated the information and structure of the project's wiki page, including all subpages.
